### ISSUE TITLE
Inconsistency in CustomerService.retrieveByEmail() And  CustomerService.create()

### ISSUE THREAD

**Issue by AlexDigital974** [medusajs/medusa#1718]
## Bug report
CustomerService.retrieveByEmail() does't work in some instance .

### Describe the bug
CustomerService.retrieveByEmail()  have  email.toLowerCase() in medusa core 
But when you create a customer Using the same service , it doesn't use a lowercase function

So you could use CustomerService.create('Test@test.com'). <--create create customer with Uppercase Email
CustomerService.retrieveByEmail('Test@test.com') <---not exist

```
    CustomerService.prototype.retrieveByEmail = function (email, config) {
        if (config === void 0) { config = {}; }
        return __awaiter(this, void 0, void 0, function () {
            var customerRepo, query, customer;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        customerRepo = this.manager_.getCustomRepository(this.customerRepository_);
                        query = this.buildQuery_({ email: email.toLowerCase() }, config);

```

### System information
irrelevant 

### Steps to reproduce the behavior

1. User CustomerService.create with customer payload and user an Uppercase Email or some uppercase stuff.
2. CustomerService.retrieveByEmail(customerEmail)  <--function will auto lowercase your input
3, See customer don't exist

### Expected behavior

If i  Create customer() with CustomerService ,i expect the same service function retrieveByEmail() to work with the same input.
Either retrieveByEmail remove the lowercase function or it is equally apply across the board.

### Extra 
Admin api  {{baseUrl}}/customers  also let uppercase in creation
{
  "email": "Test@Test.com",
  "first_name": "exercitation",
  "last_name": "elit proident",
  "phone": "cillum Lorem aute",
  "password":"TEST"
}

**Comment by srindom** [comment]
@AlexDigital974, thanks for reporting! Have opened a a PR with a fix :)

**Comment by AlexDigital974** [comment]
@srindom thanks medusa and the community for the hard work :)

**Comment by olivermrbl** [comment]
Solved in #1719
